src/d/actor/d_a_sail.cpp

```diff
@@ -449,39 +498,445 @@ void daSail_packet_c::setNrmMtx() {
 
 /* 00000398-00000424       .text setBackNrm__15daSail_packet_cFv */
 void daSail_packet_c::setBackNrm() {
-    /* Nonmatching */
+    cXyz* nrm1 = this->get_nrm1(0);
+    cXyz* nrm2 = this->get_nrm2(0);
+    for (int i = 0; i < 0x54; i++) {
+        nrm2->setall(0.0f);
+        VECSubtract(nrm2, nrm1, nrm2);
+        nrm1++;
+        nrm2++;
+    }
 }
 
 /* 00000424-00000834       .text setNrmVtx__15daSail_packet_cFP4cXyzii */
-void daSail_packet_c::setNrmVtx(cXyz*, int, int) {
-    /* Nonmatching */
+void daSail_packet_c::setNrmVtx(cXyz* param_0, int param_1, int param_2) {
+    cXyz spF8;
+    cXyz spEC;
+    cXyz spE0;
+    cXyz spD4;
+    cXyz spC8;
+    cXyz* vtxPos = this->get_pos(0);
+    s32 idx = param_2 * 7;
+    spC8.set(vtxPos[param_1 + idx]);
+    spD4.setall(0.0f);
+
+    if (param_1 != 0) {
+        spF8 = vtxPos[idx - 1 + param_1] - spC8;
+        if (param_2 != 0) {
+            spEC = vtxPos[(param_2 - 1) * 7 + param_1] - spC8;
+            spE0 = spEC.outprod(spF8);
+            spE0 = spE0.normZC();
+            VECAdd(&spD4, &spE0, &spD4);
```

Comment:
```suggestion
            spD4 += spE0;
```
also applies to other instances of VECAdd in this function